src/main/java/dev/openfeature/sdk/ImmutableContext.java

```diff
@@ -0,0 +1,123 @@
+package dev.openfeature.sdk;
+
+import lombok.Getter;
+import lombok.Setter;
+import lombok.ToString;
+import lombok.experimental.Delegate;
+
+import java.time.Instant;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * The EvaluationContext is a container for arbitrary contextual data
+ * that can be used as a basis for dynamic evaluation.
+ * The MutableContext is an EvaluationContext implementation which is not threadsafe, and whose attributes can
+ * be modified after instantiation.
+ */
+@ToString
+@SuppressWarnings("PMD.BeanMembersShouldSerialize")
+public class ImmutableContext implements EvaluationContext {
+
+    @Getter
+    private String targetingKey;
+    @Delegate(excludes = HideDelegateAddMethods.class)
+    private final MutableStructure structure;
+
+    public ImmutableContext() {
+        this.structure = new MutableStructure();
```

Comment:
I am not convinced on using `MutableStructure` for true immutability. For example, usage of `MutableStructure(Map<String, Value> attributes)` is still mutable as we do not perform a deep copy. 

I think we need an `ImutableStructure.java` implementation here. 

@toddbaert  any opinion on this ?

Reply:
I agree. I think we should create both an `ImutableStructure` an extend that for `ImmutableContext`